packages/toolpad-app/src/toolpad/propertyControls/SelectOptions.tsx

```diff
@@ -47,17 +48,43 @@ function SelectOptionsPropEditor({
     return null;
   }, [editingIndex, value]);
 
+  const switchErrorState = React.useCallback(
```

Comment:
This is a little too complicated: why create a callback with another callback as its parameter?

Reply:
> This is a little too complicated: why create a callback with another callback as its parameter?

frist all
`typeof item !== 'string' ? item.value === newOptionValue : item === newOptionValue,`
` typeof item !== 'string' ? item.value === inputText : item === inputText,`
this condition is only different in both. and the callback as parameter is easy understand